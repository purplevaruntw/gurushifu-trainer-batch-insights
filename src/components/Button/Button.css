button {
	margin-top: 100px;
	font-size: 40px;
	padding: 20px;
	background: none;
	border-radius: 9px;
	border: 4px solid black;
}
button:hover {
	border: 4px solid gold;
}
