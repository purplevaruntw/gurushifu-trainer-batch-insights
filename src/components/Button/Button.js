import React from "react";
import "./Button.css";

const Button = ({ value, onClickFunction }) => {
	return (
		<button
			value={value}
			onClick={onClickFunction}
			data-testid="batch-insights-btn"
		>
			{value}
		</button>
	);
};

export default Button;
