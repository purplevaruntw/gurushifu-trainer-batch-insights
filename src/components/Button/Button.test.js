import Button from "./Button";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

const mockFunction = jest.fn();

describe("basic rendering", () => {
	it("should render batch insights button", () => {
		const button = renderer.create(<Button value={"Batch Insights"} />);
		expect(button).toMatchSnapshot();
	});
});
describe("checking basic functionality", () => {
	it("should call onclick function when button is clicked", () => {
		render(
			<Button value={"Batch Insights"} onClickFunction={mockFunction} />
		);
		const batchInsightsButton = screen.getByTestId("batch-insights-btn");
		fireEvent.click(batchInsightsButton);
		expect(mockFunction).toBeCalled();
	});
});
