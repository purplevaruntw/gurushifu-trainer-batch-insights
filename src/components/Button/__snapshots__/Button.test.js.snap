// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering should render batch insights button 1`] = `
<button
  data-testid="batch-insights-btn"
  value="Batch Insights"
>
  Batch Insights
</button>
`;
